### PR TITLE
Email Protection sign-up tooltip

### PR DESCRIPTION
**Pull request by alistairjcbrown**
**Reviewer:** @GioSensation @shakyShane 
**Asana:** https://app.asana.com/0/0/1203490723369675/f

## Description
**Current state** -- When autofill instantiates the device interface, [it sets up the UI Controller in the constructor](https://github.com/duckduckgo/duckduckgo-autofill/blob/b16d363471003b39973b0f5418d3af2d37e657ae/src/DeviceInterface/InterfacePrototype.js#L76)

As part of this work, in the extensions, we want to show Email Signup tooltip when the user doesn't have Email Protection enabled, and Email Protection autofill when they do. This requires moving the UI Controller set up to after `setupAutofill` has completed so we know if Email Protection is enabled or not.

## Steps to test
1. Use this branch, changing the `emailProtection_incontext_signup` value in `src/Settings.js` to `true`
2. Build and use with an unpacked extension (see development steps)
3. With Email Protection disabled, go to https://spreadprivacy.com/privacy-weekly-newsletter/
4. Confirm the tootip shows with the Email Protection notice
5. Enable email protection
6. With Email Protection enabled, go to https://spreadprivacy.com/privacy-weekly-newsletter/
7. Confirm the tootip now shows with the Email Protection autofill

### CODE REVIEW

File: src/UI/EmailSignupHTMLTooltop.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        </h1>
+        <p>
+            Want me to hide your email address and remove hidden trackers before
+            forwarding messages to your inbox?
+        </p>
+        <div class="notice-controls">
+            <a href="https://duckduckgo.com/email/start-incontext" target="_blank" class="primary">
+                Get Email Protection
+            </a>
+            <button class="ghost js-dismiss-email-signup">
```
Comment (GioSensation) [inline src/UI/EmailSignupHTMLTooltop.js:25]: Lol, why the class `ghost`? Is this specified in the design language?

File: src/UI/EmailSignupHTMLTooltop.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            </button>
+        </div>
+    </div>
+</div>`
+
+        this.tooltip = this.shadow.querySelector('.tooltip')
+        this.dismissEmailSignup = this.shadow.querySelector('.js-dismiss-email-signup')
+
+        this.registerClickableButton(this.dismissEmailSignup, () => {
+            console.log('Persist dismissal')
```
Comment (GioSensation) [inline src/UI/EmailSignupHTMLTooltop.js:36]: NAB: I usually add a `TODO:` for such things so they're easier to find.

**Review by GioSensation — APPROVED** [review]
This is a great start, Alistair! The only things I'd do before merging this are:

1. Fix the "Not now" button aligning to the right (that's Chrome) 
CleanShot 2022-12-13 at 09 32 12

2. Test for regressions on all platforms

Other than that, we can merge this and keep iterating. Apart from the thing you mentioned regarding persisting the dismissal, one thing to do is automatic refresh of data so that when we get back to the initial form autofill is ready to go and no refresh is needed. We already have this elsewhere, so it should be relatively simple to hook into.

Great work so far! 💪

File: src/UI/EmailSignupHTMLTooltop.js (hunk 1 above)
Comment (alistairjcbrown) [reply src/UI/EmailSignupHTMLTooltop.js:25]: @GioSensation Yea, "ghost" is usually the name for this kind of button, where the background only shows on hover/focus (I'll need to check I've implemented that). You can see the ghost button react component on the SERP storybook

**Pull request merged**